core/src/smartcontracts/mod.rs

```diff
@@ -15,15 +15,13 @@ use crate::wsv::{WorldStateView, WorldTrait};
 pub trait Execute<W: WorldTrait> {
     /// Error type returned by execute function
     type Error: std::error::Error;
-    /// Difference between [`WorldStateView`] before and after execution of [`Self`].
-    type Diff: Into<Vec<DataEvent>>;
```

Comment:
This type is only `DataEvent` or `Vec<DataEvent>` but is always converted into `Vec<DataEvent>` except if instruction is executed on it's own, which I don't believe will ever be the case

Reply:
One more important point, is that the `DataEvent`, is not attached to the `WSV`, and is returned to the execution site. 

This opens up Iroha triggers to a WASM smartcontract just ignoring some events internally, and not returning any. It's a lot better to emit the events from the modifications of the WSV directly, and propagate them through the account/domain/world scope.

Reply:
I don't think your concern is valid. When executing instructions through wasm events are not returned to the caller. Smartcontract is unaware of and cannot accesss events

Reply:
@appetrosyan as you can see [here](https://github.com/hyperledger/iroha/blob/iroha2-dev/core/src/smartcontracts/wasm.rs#L325) or [here](https://github.com/hyperledger/iroha/pull/1894/files#diff-53286f1a9f4adfed5a98a7efba218801383f0a7b19ad3d708663a073bddb2b65R332) when executing instruction from wasm events are not returned all the way to the caller. As far as I can see, there is no possibility of attack that you are describing, it's not possible to execute instruction and not execute an event